### PR TITLE
More tests to exercise EventRouter behaviors.

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: misk/src/main/kotlin/misk/eventrouter/RealEventRouter.kt [hunk 1]
```diff
@@ -61,7 +61,7 @@ internal class RealEventRouter : EventRouter {
       if (hasClusterSnapshot.compareAndSet(false, true)) {
         this@RealEventRouter.clusterSnapshot = clusterSnapshot
         logger.debug { "[${clusterSnapshot.self}]: cluster changed: $clusterSnapshot" }
-        executor.submit({ drainQueue() })
+        executor.execute({ drainQueue() })
```
Comment (tso) [inline misk/src/main/kotlin/misk/eventrouter/RealEventRouter.kt]: Weird thought we made sure to do the right one

File: misk/src/main/kotlin/misk/eventrouter/RealEventRouterModule.kt [hunk 2]
```diff
@@ -18,9 +18,9 @@ class RealEventRouterModule : KAbstractModule() {
     install(MoshiAdapterModule(SocketEventJsonAdapter))
   }
 
-  @Provides @Singleton @ForEventRouterSubscribers
+  @Provides @Singleton @ForEventRouterActions
   fun provideExecutor(): ExecutorService =
-      ThreadPoolExecutor(5, 5, 1, TimeUnit.MINUTES, LinkedBlockingQueue())
+      ThreadPoolExecutor(1, 1, 1, TimeUnit.MINUTES, LinkedBlockingQueue())
```
Comment (tso) [inline misk/src/main/kotlin/misk/eventrouter/RealEventRouterModule.kt]: 👍

File: misk/src/test/kotlin/misk/eventrouter/EventRouterTest.kt [hunk 3]
```diff
@@ -106,6 +107,75 @@ internal class EventRouterTest {
...
+    machineA.getTopic<String>("chat").subscribe(listener)
+    processEverything()
+
+    machineA.getTopic<String>("chat").publish("message 1")
+    processEverything()
+
+    assertThat(listener.events).containsExactly("chat: open", "chat: message 1")
+  }
+
+  @Ignore("broken!")
```
Comment (tso) [inline misk/src/test/kotlin/misk/eventrouter/EventRouterTest.kt]: junit4: `@Ignore`
junit5: `@Disabled`

Comment (swankjesse) [reply misk/src/test/kotlin/misk/eventrouter/EventRouterTest.kt]: thanks

**Pull request merged**